fanctrl.py

```diff
@@ -1,146 +1,215 @@
+        try:
+            server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            server_socket.bind(COMMANDS_SOCKET_FILE_PATH)
+            os.chmod(COMMANDS_SOCKET_FILE_PATH, 0o777)
+            server_socket.listen(1)
+            while True:
+                client_socket, _ = server_socket.accept()
+                try:
+                    # Receive data from the client
+                    data = client_socket.recv(4096).decode()
+                    print("Received command:", data)
+
+                    args = parser.parse_args(shlex.split(data))
+                    if args.strategy or args._strategy:
+                        strategy = args.strategy
+                        if strategy is None:
+                            strategy = args._strategy
+                        try:
+                            if strategy == "defaultStrategy":
+                                self.clearOverwrittenStrategy()
+                            else:
+                                self.overwriteStrategy(strategy)
+                            client_socket.sendall(self.getCurrentStrategy().name.encode())
+                        except InvalidStrategyException:
+                            client_socket.sendall(("Error: unknown strategy: " + strategy).encode())
+                    if args.pause:
+                        self.pause()
+                        client_socket.sendall("Success".encode())
+                    if args.resume:
+                        self.resume()
+                        client_socket.sendall("Success".encode())
+                    if args.query:
+                        client_socket.sendall(self.getCurrentStrategy().name.encode())
+                    if args.list_strategies:
+                        client_socket.sendall('\n'.join(self.configuration.getStrategies()).encode())
+                    if args.reload:
+                        self.configuration.reload()
+                        if self.overwrittenStrategy is not None:
+                            self.overwriteStrategy(self.overwrittenStrategy.name)
+                        client_socket.sendall("Success".encode())
+                except:
+                    pass
+                finally:
+                    client_socket.shutdown(socket.SHUT_WR)
+                    client_socket.close()
+        finally:
+            server_socket.close()
 
     def setSpeed(self, speed):
         self.speed = speed
         bashCommand = f"ectool fanduty {speed}"
         subprocess.run(bashCommand, stdout=subprocess.PIPE, shell=True)
 
-    def adaptSpeed(self):
-        currentTemp = self.temps[self._tempIndex]
-        currentTemp = min(
-            currentTemp, self.getMovingAverageTemperature(self.movingAverageInterval)
-        )
-        minPoint = self.speedCurve[0]
-        maxPoint = self.speedCurve[-1]
-        for e in self.speedCurve:
+    def getActualTemperature(self):
+        bashCommand = "ectool temps all"
```

Comment:
I don't think we should go this way, averaging all temperatures reported by ectool.
For instance, with this code, if the CPU goes from 40C to 80C in a few seconds, then this function will only return a temperature increase of 40C/N (where N is the number of entries returned by `ectool temps all`
It means that fw-fanctrl will not react much, until the cpu heat reaches other components so the average temp increases. Before that the CPU will remain super hot, which isn't too good for it. Same reasoning apply to GPU.

On the other hand, current algorithm doesn't take into account the other temperature sensors, only CPU and GPU. Adding support for battery  and others is not trivial since they have different temperature tolerances than CPU/GPU.

I feel like current algorithm is good enough. You might change it to use temp inputs from `ectool` only, though.
What do you think ?